fast64_internal/f3d/f3d_bleed.py

```diff
@@ -243,12 +263,45 @@ def bleed_mat(self, cur_fmat: FMaterial, last_mat: FMaterial, bleed_state: int):
             commands_bled = copy.copy(gfx)
             commands_bled.commands = copy.copy(gfx.commands)  # copy the commands also
             last_cmd_list = last_mat.mat_only_DL.commands
+
+            # handle write diff, save pre bleed cmds
+            is_rendermode_cmd = lambda cmd: isinstance(cmd, DPSetRenderMode) or (
+                isinstance(cmd, SPSetOtherMode) and cmd.sets_rendermode
+            )
+            geo_cmd = next((cmd for cmd in commands_bled.commands if type(cmd) == SPGeometryMode), None)
+            othermode_cmds = [cmd for cmd in commands_bled.commands if isinstance(cmd, SPSetOtherModeSub)]
+            rendermode_cmds = [cmd for cmd in commands_bled.commands if is_rendermode_cmd(cmd)]
+
             for j, cmd in enumerate(gfx.commands):
                 if self.bleed_individual_cmd(commands_bled, cmd, bleed_state, last_cmd_list):
                     commands_bled.commands[j] = None
             # remove Nones from list
             while None in commands_bled.commands:
                 commands_bled.commands.remove(None)
+
```

Comment:
I'm not exactly sure what, but something funky is going on here. I inspected some display lists exported using bleed and they were not properly setting the render modes.
the test in question was to have two materials that are the same, but with different geo/render modes. the geo modes were properly set to match the material, but the render modes were not. In fact the render mode setting for the second material was completely missing